dev/gluten-build-info.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+VELOX_HOME="$1"
```

Comment:
Is it more appropriate to use BACKEND BACKEND_TYPE and BACKEND_HOME to compatible with clickhouse backend?

Reply:
Hi @lwz9103, I am not familiar with CH backend's build process. For velox backend, we clone a velox repo to a local directory inside gluten home. So with the proposed patch, we can go into it to get required version info. Does CH do the similar build process? If so, please let the author know the relative path for CH local repo.

Reply:
I think it makes sense. The reason I only do it for velox is I'm not familiar with ch backend and also do not have a env for ch backend... @lwz9103  can you make a followup after this pr merged ? thank you !

Reply:
@PHILO-HE Yes,  in the process of building CH, BUILD_INFO will be also generated, but the content is slightly different.  I will try to make the generation  and basic information of build_info in CH bakend consistent with velox backend.

Reply:
@ulysses-you It's my pleasure to do it